File: php/class/user.class.php
<?php
class User
{
    public int $id;
    public string $nom;
    public string $prenom;
    public string $email;

    public function __construct($id = 42, $nom = "SMITH", $prenom = "John", $email = "noreply@ici.com")
    {
        $this->id = $id;
        $this->nom = $nom;
        $this->prenom = $prenom;
        $this->email = $email;
    }

    public function display()
    {
        return "" . $this->id . "\t" . $this->nom . "\t" . $this->prenom . "\t" . $this->email . "\n";
    }
}


// $u = new User();
// var_dump($u);
// $u->id = 22;
// $u->nom = "Tata";
// $u->prenom = "Yoyo";
// $u->email = "tata@yoyo.com";
// var_dump($u);

$u2 = new User(5, "truc 2", "machin");
echo ($u2->display());

File: Python/classes/user.class.py
class User:
    id: int
    nom: str
    prenom: str
    email: str

    def __init__(self, id=42, nom="SMITH", prenom="John", email="noreply@ici.com"):
        self.id = id
        self.nom = nom
        self.prenom = prenom
        self.email = email

    def display(self):
        return (
            ""
            + str(self.id)
            + "\t"
            + self.nom
            + "\t"
            + self.prenom
            + "\t"
            + self.email
            + "\n"
        )


u = User(id=4, email="test@test.com")
print(u.display())
